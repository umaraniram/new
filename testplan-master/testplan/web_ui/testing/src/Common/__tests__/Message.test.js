import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import Message from '../Message';

function defaultProps() {
  return {
    left: '1.5em',
    message: 'test',
  };
}

describe('Message', () => {
  let props;
  let mountedMessage;
  const renderMessage = () => {
    if (!mountedMessage) {
      mountedMessage = shallow(
        <Message {...props} />
      );
    }
    return mountedMessage;
  };

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    mountedMessage = undefined;
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('shallow renders without crashing', () => {
    renderMessage();
  });

  it('shallow renders the correct HTML structure', () => {
    const message = renderMessage();
    expect(message).toMatchSnapshot();
  });

});