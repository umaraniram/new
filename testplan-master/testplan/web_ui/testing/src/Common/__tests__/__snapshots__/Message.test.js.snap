// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message shallow renders the correct HTML structure 1`] = `
<div
  style={
    Object {
      "paddingLeft": "1.5em",
      "paddingTop": "4.5em",
    }
  }
>
  <h1
    className="message_110xrpz"
  >
    test
  </h1>
</div>
`;
